system/Config/BaseConfig.php

```diff
@@ -103,6 +103,9 @@ protected function initEnvValue(&$property, string $name, string $prefix, string
                 $value = true;
             }
             $property = is_bool($value) ? $value : trim($value, '\'"');
+            if (preg_match("/^[0-9]+$/", $property)) {
+                $property = (int) $property;
+            }
```

Comment:
Placing this after the trim removes the possibility of passing string numerals. Maybe switch the order of these? Also, any sense to expanding this to `is_numeric()` and casting to float more generally?